web500/routes.py

```diff
@@ -20,11 +32,10 @@ def login():
         return redirect(url_for("index"))
 
     if request.method == "POST":
-        if not query_db("select * from users where username = ?",
-                        [request.form['username']]):
-            session['username'] = request.form['username']
-            update_db("insert into users (username) values (?)",
-                      [request.form['username']])
+        username = request.form['username']
+        if _is_valid(username) and not query_db("select * from users where username = ?", [username]):
```

Comment:
We will most certainly rewrite this line when we change the username structure though.